### PR TITLE
feat: add extra derives option

### PR DESCRIPTION
**Pull request by ouertani**
Allow the user to define his own extra derives, which will provide more flexibility to customize the generated structs.

### CODE REVIEW

**Comment by lerouxrgd** [comment]
This looks good ! Could you also provide a few tests for this feature ?

**Comment by ouertani** [comment]
@lerouxrgd i added a couple of tests

File: README.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ Options:
       --chrono-dates     Use chrono::NaiveDateTime for date/timestamps logical types
       --derive-builders  Derive builders for generated record structs
       --derive-schemas   Derive AvroSchema for generated record structs
+      --extra_derive    Append one extra derive macros to the generated record structs
```
Comment (martin-g) [inline README.md:35]:
```suggestion
      --extra_derive     Append one extra derive macros to the generated record structs
```

Comment (martin-g) [reply README.md:35]: IMO there is no need of `--extra_derive`.
`--extra_derives` should be enough for everyone.

File: src/main.rs [hunk 2]
```diff
@@ -44,6 +44,10 @@ struct Args {
     /// Derive AvroSchema for generated record structs
     #[clap(long)]
     pub derive_schemas: bool,
+
+    /// Extract Derives for generated record structs
+    #[clap(long)]
+    pub extra_derives: Vec<String>,
```
Comment (martin-g) [inline src/main.rs:50]: There is no option for `--extra_derive`. 
As I said earlier it is not really needed but I wonder how it is supposed to work now.

File: src/main.rs [hunk 3]
```diff
@@ -44,6 +44,10 @@ struct Args {
     /// Derive AvroSchema for generated record structs
     #[clap(long)]
     pub derive_schemas: bool,
+
+    /// Extract Derives for generated record structs
```
Comment (martin-g) [inline src/main.rs:48]:
```suggestion
    /// Extract Derives for generated record structs, comma separate, e.g. `...`
```
?

**Comment by lerouxrgd** [comment]
Thanks for the PR !

**Pull request merged**